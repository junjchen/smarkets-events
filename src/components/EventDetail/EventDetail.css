.event-detail__name {
    font-weight: 400
}

.event-detail__link {
    display: block;
    margin: 1em 0;
}