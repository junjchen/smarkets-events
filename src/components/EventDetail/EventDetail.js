import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import moment from 'moment'
import StatusBar from '../StatusBar'
import './EventDetail.css'

class EventDetail extends Component {
    componentDidMount() {
        const {fetchData, id} = this.props
        fetchData(id)
    }
    render() {
        const {event, state} = this.props
        return (
            <section className="event-detail">
                <StatusBar {...this.props}/> {!!event && (
                    <div>
                        <h1 className="event-detail__name">{event.name}
                            ({event.short_name})</h1>
                        <p className="event-detail__category">{event.parent_name}</p>
                        <p className="event-detail__date">{moment(event.start_datetime).format('MMMM Do YYYY, h:mm:ss a')}</p>
                    </div>
                )}
                <Link className="event-detail__link" to="/popular">Go back</Link>
            </section>
        )
    }
}

export default EventDetail