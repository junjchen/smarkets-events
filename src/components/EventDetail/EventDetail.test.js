import React from 'react'
import {shallow, mount} from 'enzyme'
import EventDetail from './EventDetail'
import moment from 'moment'
import {MemoryRouter} from 'react-router-dom'

const createProps = () => ({
    id: 1,
    event: {
        id: 1,
        name: 'name',
        short_name: 'short_name',
        parent_name: 'category',
        start_datetime: new Date('1990/10/10')
    },
    fetchData: jest.fn()
})

describe('EventDetail', () => {
    it('renders same as snapshot', () => {
        const props = createProps()
        const wrapper = shallow(<EventDetail {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })
    it('calls fetchData', () => {
        const props = createProps()
        const wrapper = mount(
            <MemoryRouter>
                <EventDetail {...props}/>
            </MemoryRouter>
        )
        expect(props.fetchData).toBeCalled()
    })
})
