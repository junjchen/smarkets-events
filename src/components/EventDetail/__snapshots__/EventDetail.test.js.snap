// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetail renders same as snapshot 1`] = `
<section
  className="event-detail"
>
  <StatusBar
    event={
      Object {
        "id": 1,
        "name": "name",
        "parent_name": "category",
        "short_name": "short_name",
        "start_datetime": 1990-10-09T23:00:00.000Z,
      }
    }
    fetchData={[Function]}
    id={1}
  />
   
  <div>
    <h1
      className="event-detail__name"
    >
      name
      (
      short_name
      )
    </h1>
    <p
      className="event-detail__category"
    >
      category
    </p>
    <p
      className="event-detail__date"
    >
      October 10th 1990, 12:00:00 am
    </p>
  </div>
  <Link
    className="event-detail__link"
    replace={false}
    to="/popular"
  >
    Go back
  </Link>
</section>
`;
